src/components/v5/common/ActionSidebar/partials/Motions/steps/StakingStep/partials/StakingForm/StakingForm.tsx

```diff
@@ -0,0 +1,204 @@
+                name="voteType"
+                items={STAKING_RADIO_BUTTONS}
+                disabled={disableForm}
+              />
+            </div>
+            <AnimatePresence>
+              {voteTypeValue !== undefined && (
+                <motion.div
+                  key="accordion-content"
+                  initial="hidden"
+                  animate="visible"
+                  exit="hidden"
+                  variants={accordionAnimation}
+                  transition={{ duration: 0.4, ease: 'easeOut' }}
+                  className="mt-6"
+                >
+                  <label
+                    htmlFor="amount-field"
+                    className="flex justify-between items-center gap-x-4 flex-wrap mb-2"
+                  >
+                    <span className="text-1 text-gray-900">
+                      {formatText({ id: 'motion.staking.input.label' })}
+                    </span>
+                    <span className="text-sm text-gray-600">
+                      {formatText(
+                        { id: 'motion.staking.input.balance' },
+                        {
+                          balance: (
+                            <Numeral
+                              value={userActivatedTokens}
+                              suffix={tokenSymbol}
+                              decimals={tokenDecimals}
+                            />
+                          ),
+                        },
+                      )}
+                    </span>
+                  </label>
+                  <FormFormattedInput
+                    id="amount-field"
+                    name="amount"
+                    options={{
+                      numeral: true,
+                      numeralDecimalScale:
+                        getTokenDecimalsWithFallback(tokenDecimals),
+                      numeralPositiveOnly: true,
+                      rawValueTrimPrefix: true,
+                      prefix: tokenSymbol,
+                      tailPrefix: true,
+                    }}
+                    buttonProps={{
+                      label: formatText({ id: 'button.max' }) || '',
+                      onClick: () => {
+                        setValue(
+                          'amount',
+                          voteTypeValue === MotionVote.Yay
+                            ? moveDecimal(
+                                supportRemaining,
+                                -getTokenDecimalsWithFallback(tokenDecimals),
+                              )
+                            : moveDecimal(
+                                opposeRemaining,
+                                -getTokenDecimalsWithFallback(tokenDecimals),
+                              ),
+                          {
+                            shouldTouch: true,
+                            shouldValidate: true,
+                            shouldDirty: true,
+                          },
+                        );
```

Comment:
can you extract this to hook or util?

Reply:
I don't see a point in doing so. I would have to pass everything inside, because I can't use form context hook inside, because it won't work